src/boot/bin/XMLToSARIF.sh

```diff
@@ -87,43 +83,33 @@ if [ -z "$xml_report_path" ]; then
     echo "Error: \"-i\" or \"--inputXmlReport\" is required."
     print_usage
     exit 1
-elif [ ! -f "$xml_report_path" ]; then
-    echo "Error: \"$xml_report_path\" not found."
-    exit 1
-fi
-
-# Set default SARIF report path if not provided
-if [ -z "$sarif_report_path" ]; then
-    sarif_report_path="${xml_report_path%.*}.sarif"
 fi
 
-# Find Java path
+# Find Java path and set temporary JAVA_HOME if necessary
 if [ -z "$parasoft_tool_or_java_root_path" ]; then
     java_path=$(which java)
     if [ -z "$java_path" ]; then
         echo "Error: Java is not found. Please add Java in environment variable or provide \"-t\" or \"--toolOrJavaHomeDir\" option."
         exit 1;
     fi
 else
-    if results=$(getJavaPath "$parasoft_tool_or_java_root_path"); then
-        java_path="$results"
+    if java_path=$(getJavaPath "$parasoft_tool_or_java_root_path"); then
+        java_home="${java_path%bin/java}"
+        export JAVA_HOME=$java_home
+        echo "Using temporary JAVA_HOME: $JAVA_HOME"
     else
         echo "Error: Tool or Java home directory is incorrect: \"$parasoft_tool_or_java_root_path\". Please check \"-t\" or \"--toolOrJavaHomeDir\" value."
         exit 1;
     fi
 fi
 
 ## 4. Generate SARIF report
+args=(
+  -i "$xml_report_path"
+)
+if [ -n "$sarif_report_path" ]; then
+  args+=(-o "$sarif_report_path")
+fi
 
-command="\"$java_path\" -jar \"$BASE_DIR/lib/saxon/saxon-he-12.2.jar\" -xsl:\"$SARIF_XSL_PATH\" -s:\"$xml_report_path\" -o:\"$sarif_report_path\" "
-eval "$command";
+./parasoft-report-transformer xml2sarif "${args[@]}"
```

Comment:
I think we should keep BASE_DIR and use absolute path here for parasoft-report-transformer.
I meet this error when i tried to call this script in another directory like: **./bin/XMLToSARIF.sh**
`./bin/XMLToSARIF.sh: line 114: ./parasoft-report-transformer: No such file or directory`